src/runtime/rpc/rpc_endpoint.cc

```diff
@@ -700,6 +700,14 @@ void RPCEndpoint::Init() {
   });
 }
 
+/*!
+ * \brief Create a new RPCEndpoint instance.
+ * \param channel RPCChannel used to communicate
+ * \param name
+ * \param The remote key reported during protocol initialization, or "%toinit" if the RPCEndpoint
+ *     should handle this phase of the protocol for you. Some servers may prefer to access parts of
+ *     the key to modify their behavior.
+ */
```

Comment:
Related to the RPC mechanism, why isn't this an independent PR ?

Reply:
i'm just adding documentation where it was helpful as I was reading the code. happy to split it out; the regression takes long.